src/PaymentGateways/Gateways/Stripe/Actions/CreatePaymentIntent.php

```diff
@@ -53,50 +46,50 @@ public function __invoke(
         $intent_args = apply_filters(
             'give_stripe_create_intent_args',
             array_merge([
-                'amount' => Money::of($paymentData->price, $paymentData->currency)->getMinorAmount(),
-                'currency' => $paymentData->currency,
+                'amount' => $donation->amount->formatToMinorAmount(),
+                'currency' => $donation->amount->getCurrency(),
                 'payment_method_types' => ['card'],
                 'statement_descriptor' => give_stripe_get_statement_descriptor(),
                 'description' => $donationSummary->getSummaryWithDonor(),
-                'metadata' => give_stripe_prepare_metadata(
-                    $paymentData->donationId,
-                    new LegacyDonationData($paymentData, $paymentMethod->id())
-                ),
+                'metadata' => give_stripe_prepare_metadata($donation->id),
                 'customer' => $giveStripeCustomer->get_id(),
                 'payment_method' => $paymentMethod->id(),
                 'confirm' => true,
-                'return_url' => $paymentData->redirectUrl,
+                'return_url' => give_get_success_page_uri(),
```

Comment:
I want to think through this a bit more. Is this how gateways tell the Gateway API they're done?

Reply:
@JasonTheAdams are you thinking there should be some kind of route that gateways return to after success and then get redirected to the success page?

Reply:
I just want to think through it, honestly. It's probably fine for now. Basically, some forms have a receipt, so they just go back to the same page; other forms do direct the user to a distinct receipt page. I just want to make sure this flexibility is accounted for. It looks like that function just returns the URL of the distinct receipt page. So does this work in multi-step?

Reply:
I think that function runs through a filter and then the page figures out where to direct them.  @ravinderk would know best.

Reply:
@JasonTheAdams Yes, the multi-step form template uses filter to change receipt page URL to redirect the donor to the correct receipt page.

Reply:
And it's probably doing it based on the request superglobals, right? The function is a reasonable abstraction, but I'm a little worried about it down the road. @jonwaldstein Do you think this is fine for where things are going in Next Gen?